File: Algorithms/src/1161. Maximum Level Sum of a Binary Tree/Code/solution.java
class Solution {
    public int maxLevelSum(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        int level = 1;
        int answerLevel = 1;
        long maxSum = Long.MIN_VALUE;

        while (!queue.isEmpty()) {
            int size = queue.size();
            long levelSum = 0;

            for (int i = 0; i < size; i++) {
                TreeNode node = queue.poll();
                levelSum += node.val;

                if (node.left != null) queue.add(node.left);
                if (node.right != null) queue.add(node.right);
            }

            if (levelSum > maxSum) {
                maxSum = levelSum;
                answerLevel = level;
            }

            level++;
        }

        return answerLevel;
    }
}

File: Algorithms/src/1161. Maximum Level Sum of a Binary Tree/Code/solution.js
var maxLevelSum = function (root) {
  let queue = [root];
  let level = 1;
  let answerLevel = 1;
  let maxSum = -Infinity;

  while (queue.length > 0) {
    let size = queue.length;
    let levelSum = 0;

    for (let i = 0; i < size; i++) {
      let node = queue.shift();
      levelSum += node.val;

      if (node.left) queue.push(node.left);
      if (node.right) queue.push(node.right);
    }

    if (levelSum > maxSum) {
      maxSum = levelSum;
      answerLevel = level;
    }

    level++;
  }

  return answerLevel;
};
